src/publisher.ts

```diff
@@ -157,17 +154,21 @@ export class Publisher {
    * //-
    * publisher.publish(data).then((messageId) => {});
    */
-  publish(data: Buffer, attributes?: object): Promise<string>;
+  publish(data: Buffer, attributes?: Attributes): Promise<string>;
   publish(data: Buffer, callback: PublishCallback): void;
-  publish(data: Buffer, attributes: object, callback: PublishCallback): void;
-  publish(data: Buffer, attributes?, callback?): Promise<string>|void {
+  publish(data: Buffer, attributes: Attributes, callback: PublishCallback):
+      void;
+  publish(
+      data: Buffer, attributesOrCallback?: Attributes|PublishCallback,
+      callback?: PublishCallback): Promise<string>|void {
     if (!(data instanceof Buffer)) {
       throw new TypeError('Data must be in the form of a Buffer.');
     }
-    if (is.fn(attributes)) {
-      callback = attributes;
-      attributes = {};
-    }
+    const attributes =
```

Comment:
What version of `@google-cloud/common` is tied to this? Can we use [maybeOptionsOrCallback](https://github.com/googleapis/nodejs-common/blob/master/src/util.ts#L756)?

Reply:
nodejs-pubsub is not using @google-cloud/common.